superduperdb/backends/ibis/query.py

```diff
@@ -492,10 +519,43 @@ def _outputs(self, query_id: str, **kwargs):
             attr = getattr(  # type: ignore[call-overload]
                 self, self.table_or_collection.primary_id
             )
-            other_query = self.join(symbol_table, symbol_table.input_id == attr)
+            other_query = self.join(symbol_table, symbol_table._input_id == attr)
             other_query = other_query.filter(other_query.key == key)
             return other_query
 
+    def __call__(self, *args, **kwargs):
+        primary_id = (
+            [self.primary_id]
+            if isinstance(self.primary_id, str)
+            else self.primary_id[:]
+        )
+
+        for a in args:
+            if isinstance(a, IbisQueryLinker) or isinstance(a, IbisQueryTable):
+                pid = [a.primary_id] if isinstance(a.primary_id, str) else a.primary_id
+                primary_id.extend(pid)
+
+        for v in kwargs.values():
+            if isinstance(v, IbisQueryLinker) or isinstance(v, IbisQueryTable):
+                pid = [v.primary_id] if isinstance(v.primary_id, str) else v.primary_id
+                primary_id.extend(pid)
+
+        primary_id = [p for p in primary_id if p != '_input_id']
```

Comment:
why would `p` be _input_id?

Reply:
This is now the name of `primary_id` of a `Model` table.